github_connector/models/github_repository_branch.py

```diff
@@ -238,21 +242,71 @@ def _get_analysis_rules(self):
             return self.analysis_rule_ids
         return self.repository_id._get_analysis_rules() + self.analysis_rule_ids
 
-    def set_analysis_rule_info(self):
-        for rule_id in self._get_analysis_rules():
-            self._process_analysis_rule_info(rule_id)
+    def _call_cloc_command(self):
+        """Execute the cloc command and save the result temporarily to access it in
+        multiple places."""
+        return check_output(["cloc", "--by-file", "--json", self.local_path])
 
-    def _process_analysis_rule_info(self, rule_id):
+    def set_analysis_rule_info(self):
+        """Do the cloc command only once and perform the analysis process."""
+        rules = self._get_analysis_rules()
+        if rules:
+            self.cloc_response = self._call_cloc_command()
```

Comment:
Why not passing the response as argument instead? This technique is a bit risky due to possible invalidation / worker respawn

Reply:
I haven't used that technique because in `github_connector_odoo` you would have to pass that argument in many methods and it would look a bit ugly, but it's not a problem and it could be done, do you think it would be better and we remove `cloc_response` field?

Reply:
Changes done (`cloc_response` field has been removed)

Reply:
Yes, although it may seem ugly at the beginning, it's needed for avoiding side effects, avoiding the [global variable problems](https://www.baeldung.com/cs/global-variables) and also invalidations of the models in between, emptying the field content.